sunpy/net/helioviewer.py

```diff
@@ -88,7 +81,7 @@ def get_closest_image(self, date, **kwargs):
 
         return response
 
-    def download_jp2(self, date, directory=None, overwrite=False, **kwargs):
+    def download_jp2(self, date, sourceId, directory=None, overwrite=False, **kwargs):
```

Comment:
So now, do we still need `**kwargs`?

Reply:
We can use **kwargs as we are still passing keyworded variable length of arguments

Reply:
Are we? What extra kwargs can we pass in now?

Reply:
But aren't we updating the parameters using kwargs?

Reply:
`**kwargs` allows a user to pass in extra arguments. It allowed the old optional arguments to be passed in without having to be defined.

Reply:
I will test it
I think we should keep it

Reply:
DId you see if we needed this?

Reply:
Yeah we need them as we are passing some extra parameters in some of the examples

Reply:
Either we have to change all the example or we need **kwargs

Reply:
Well those `kwargs` do anything? If not,  they need to be removed from the examples.

Reply:
It is like we will be able to pass extra arguments without defining them though we only need sourceId and date but if anyone wishes to add extra arguments with them we will be able to

Reply:
But we shouldn't allow that unless there is a reason.

Reply:
Ok then there is no need for that

Reply:
I guess we need it because of this
 # JPIP URL response
        if 'jpip' in kwargs:
            return self._get_json(params)

        return self._get_file(params, directory, overwrite=overwrite)

Reply:
And we don't even need to describe the optional parameters if we won't let the users use them?

Reply:
Ah I see now. But what about `json`? I don't see where that gets handled in the code. 

Yeah, if we don't support extra keywords, we don't need to describe them.

Reply:
I guess there is some confusion in the documentation of Helioviewer API
The takeScreenshot function says that we must specify 
 either `x1`, `y1`, `x2`, and `y2`
or `x0`, `y0`, `width` and `height`.
But if we see the required parameters they are optional

Reply:
> Ah I see now. But what about `json`? I don't see where that gets handled in the code.
> 
> Yeah, if we don't support extra keywords, we don't need to describe them.

We dont need anything for json but yes we do need something if both jpip and json are set to true.

Reply:
Well we have to make those changes too then. We should match the official API where possible.

You should check to see if the official API can handle both json and jpip are set to True to make sure it works.  Otherwise we need to work around that.

Reply:
It works in the official API. We have to provide that feature then. I will see what all I can do

Reply:
> I guess there is some confusion in the documentation of Helioviewer API
> The takeScreenshot function says that we must specify
> either `x1`, `y1`, `x2`, and `y2`
> or `x0`, `y0`, `width` and `height`.
> But if we see the required parameters they are optional

What about this? What shall I do??

Reply:
What do you mean?

Reply:
The actual documentation of Helioviewer API says 
The takeScreenshot function says that we must specify
either x1, y1, x2, and y2
or x0, y0, width and height.

But it is not a required parameter

Reply:
You mean in our code? Then we should make it one?

Reply:
No no 
This documentation says this 
https://api.helioviewer.org/docs/v2/

Reply:
I wonder what returns if you don't supply those keywords, it must be the full disc. 

Well if they are optional, then nothing changes?

Reply:
Yup but the thing is the documentation says you must specify these parameters 
So we have to keep **kwargs because of this

Reply:
Sure but isn't that another function?

Reply:
Yup the download_png function but I was confused

Reply:
Nope cause the documentation says you must specify them even though they are not required

Reply:
Does this function call the other function?

Reply:
Sorry I didn't get that

Reply:
There are two independent functions using **kwargs as their argument. The thing is we cannot remove them from any function.

Reply:
Why can we not from this function?

Reply:
By this you mean the download_png function??

Reply:
No the `download_jp2` function?

Reply:
JPIP URL response

    if 'jpip' in kwargs:
        return self._get_json(params)

    return self._get_file(params, directory, overwrite=overwrite)
Because of this check

Reply:
Ah I missed that. I don't see why it can't be made into a proper keyword argument tho and just set to false.

Reply:
If you want I can do that but the API lets you set it to true

Reply:
Which means its false by default. But we've gone off topic. We can leave it the way it is for now.

Reply:
Ok I will try that